test/validation/test_formatting_validation.py

```diff
@@ -345,6 +346,23 @@ def test_check_for_invalid_header(test_input: str, should_match: bool) -> None:
     assert actual == expected
 
 
+@pytest.mark.parametrize(
+    "test_input,should_match",
+    [
+        ("[lance]: https://en.wikipedia.org/wiki/Holy_Lance \"Holy Lance - Wikipedia\"", True),
+        ("[**Unknown User**]: Oh man i think i just ran out of pain", True),
+        ("something something something [**Unknown User**]: Oh man i think i just ran out of pain", True),
+        ("[**Unknown User**]:Oh man i think i just ran out of pain", True),
+        (r"\[**Unknown User**]: Oh man i think i just ran out of pain", False)
+    ]

```

Comment:
Let's add a test here that a normal description with `[*Bla bla bla description*]` does not trigger this check.

I think we'll have to be careful that this check doesn't add too many false positives, but if necessary we can also relax it later.

Reply:
The description test will be added in the soon-to-be commits. I can see how this could cause some false-positives but it could be minimized by only having the regex trigger at the start of lines. From my testing, the reference link formatting only works at the start of the line although I only did very basic testing on desktop old reddit, it could be different elsewhere.